fairmq/FairMQDevice.h

```diff
@@ -338,6 +332,7 @@ class FairMQDevice : public FairMQStateMachine, public FairMQConfigurable
 
     /// Creates stand-alone transport factory
     /// @param transport  Transport string ("zeromq"/"nanomsg"/"shmem")
+    [[deprecated("use 'auto FairMQTransportFactory::CreateTransportFactory(const std::string& type) -> std::shared_ptr<FairMQTransportFactory>' instead")]]
     static std::unique_ptr<FairMQTransportFactory> MakeTransport(const std::string& transport);
```

Comment:
We found that this function was still used currently in AliceO2 code, so we just marked it deprecated. This `[[deprecate("msg")]]` attribute is standardized in C++14, but the attribute syntax `[[ ]]` was already introduced in [C++11](http://en.cppreference.com/w/cpp/language/attributes). Worst case should be, that a C++11 compiler will just ignore it (which is only guaranteed in C++17 standard though).